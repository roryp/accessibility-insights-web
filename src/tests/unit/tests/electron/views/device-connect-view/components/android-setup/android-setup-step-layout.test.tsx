// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    AndroidSetupStepLayout,
    AndroidSetupStepLayoutProps,
} from 'electron/views/device-connect-view/components/android-setup/android-setup-step-layout';
import { shallow } from 'enzyme';
import * as React from 'react';

describe('AndroidSetupStepLayout', () => {
    let props: AndroidSetupStepLayoutProps;
    const stubChildContent = <p>child content</p>;
    beforeEach(() => {
        props = {
            leftFooterButtonProps: { text: 'left footer button' },
            rightFooterButtonProps: { text: 'right footer button' },
        };
    });

    it.each`
        withMoreInfo | withHeaderText
        ${true}      | ${true}
        ${false}     | ${true}
        ${false}     | ${false}
    `(
        'renders per snapshot (withMoreInfo=$withMoreInfo, withHeaderText=$withHeaderText)',
        ({ withMoreInfo, withHeaderText }) => {
            props.headerText = withHeaderText ? 'Header text' : undefined;
            props.moreInfoLink = withMoreInfo ? <a href="https://test">test link</a> : undefined;

            const rendered = shallow(
                <AndroidSetupStepLayout {...props}>{stubChildContent}</AndroidSetupStepLayout>,
            );
            expect(rendered.getElement()).toMatchSnapshot();
        },
    );
});
