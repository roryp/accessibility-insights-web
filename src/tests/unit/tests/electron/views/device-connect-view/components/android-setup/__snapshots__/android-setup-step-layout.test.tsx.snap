// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AndroidSetupStepLayout renders per snapshot (withMoreInfo=false, withHeaderText=false) 1`] = `
<main
  className="layoutContainer"
>
  <div
    className="content"
  >
    <p>
      child content
    </p>
  </div>
  <div
    className="footer"
  >
    <CustomizedDefaultButton
      data-automation-id="android-left-footer-button"
      text="left footer button"
    />
    <CustomizedPrimaryButton
      data-automation-id="android-right-footer-button"
      text="right footer button"
    />
  </div>
</main>
`;

exports[`AndroidSetupStepLayout renders per snapshot (withMoreInfo=false, withHeaderText=true) 1`] = `
<main
  className="layoutContainer"
>
  <h1
    className="header"
  >
    Header text
  </h1>
  <div
    className="content"
  >
    <p>
      child content
    </p>
  </div>
  <div
    className="footer"
  >
    <CustomizedDefaultButton
      data-automation-id="android-left-footer-button"
      text="left footer button"
    />
    <CustomizedPrimaryButton
      data-automation-id="android-right-footer-button"
      text="right footer button"
    />
  </div>
</main>
`;

exports[`AndroidSetupStepLayout renders per snapshot (withMoreInfo=true, withHeaderText=true) 1`] = `
<main
  className="layoutContainer"
>
  <h1
    className="header"
  >
    Header text
  </h1>
  <div
    className="moreInfoLink"
    data-automation-id="more-info-link"
  >
    <a
      href="https://test"
    >
      test link
    </a>
  </div>
  <div
    className="content"
  >
    <p>
      child content
    </p>
  </div>
  <div
    className="footer"
  >
    <CustomizedDefaultButton
      data-automation-id="android-left-footer-button"
      text="left footer button"
    />
    <CustomizedPrimaryButton
      data-automation-id="android-right-footer-button"
      text="right footer button"
    />
  </div>
</main>
`;
